lua/orgmode/files/headline.lua

```diff
@@ -113,8 +113,8 @@ function Headline:demote(amount, recursive, dryRun)
       if line:sub(1, 1) == '*' then
         lines[i] = string.rep('*', amount) .. line
       else
-        if config.org_adapt_indentation then
-          lines[i] = config:apply_indent(line, amount)
+        if config:should_indent(self.file:bufnr()) then
+          lines[i] = self:_apply_indent(line, amount)
         else
           line, _ = line:gsub('^%s+', '')
           local indent_amount = indent.indentexpr(start_line + i)
```

Comment:
Add:
```suggestion
          local indent_amount = indent.indentexpr(start_line + i, self.file:bufnr())
```
to take advantage of the new `bufnr` arg on the `indentexpr` function.

Reply:
Added it, thanks.